lib/dialects/postgres/index.js

```diff
@@ -105,7 +106,8 @@ Object.assign(Client_PG.prototype, {
   // connection needs to be added to the pool.
   acquireRawConnection() {
     const client = this;
-    return new Bluebird(function(resolver, rejecter) {
+    return new Bluebird(async function(resolver, rejecter) {
+      await client.connection().resolveConnectionConfig();
```

Comment:
We can simplify this, I think, instead of 
```
await client.connection().resolveConnectionConfig();
const connection = new client.driver.Client(client.connectionSettings);
```

We can do just
```
const connection = await client.connection().resolveConnectionConfig();
```

Reply:
In case we need connectionSettings anywhere else, we could also add `client.connectionSettings = connectionConfig` here, but it shouldn't be provider's responsibility to mutate that state.

Reply:
As requested
```
const connectionSettings = await client
        .connectionConfigProvider()
        .resolveConnectionConfig();
      const connection = new client.driver.Client(connectionSettings);
```
https://github.com/tgriesser/knex/pull/3403/files#diff-0fdffe855b98a6c04d603ff6c442c163R110

I agree with you that we shouldn't mutate the state of Client. but if `client.connectionSettings ` is used somewhere else, then this might cause a problem.

What I have done is I mutated the state of the client in `connectionProvider.js`
https://github.com/tgriesser/knex/pull/3403/files#diff-cbc1ae6887774cf34c8d4bc73dece12fR14